pkg/storegateway/bucket_stores.go

```diff
@@ -551,17 +559,31 @@ func (u *BucketStores) closeBucketStoreAndDeleteLocalFilesForExcludedTenants(inc
 	}
 }
 
-// getBlocksLoadedMetric returns the number of blocks currently loaded across all bucket stores.
-func (u *BucketStores) getBlocksLoadedMetric() float64 {
-	count := 0
+func (u *BucketStores) Describe(descs chan<- *prometheus.Desc) {
+	descs <- u.blocksLoaded
+	descs <- u.blocksLoadedByDuration
+}
+
+func (u *BucketStores) Collect(metrics chan<- prometheus.Metric) {
+	loaded := make(map[time.Duration]int)
+	total := 0
 
 	u.storesMu.RLock()
+
 	for _, store := range u.stores {
-		count += store.Stats().BlocksLoaded
+		stats := store.Stats(u.cfg.TSDB.BlockRanges)
+		for d, n := range stats.BlocksLoaded {
+			loaded[d] += n
+			total += n
+		}
 	}
+
 	u.storesMu.RUnlock()
```

Comment:
Super-nit: Can we extract this into a function and use `defer` for unlock? It makes it easier to see and verify that locking is correct here.

Reply:
Sure, I'll see if I can split this out.

Reply:
Moved the summing of counts and locking to a separate method in 8fa952c172f6028d6be259f3c5aeb7da7f4bc795